fswatcher/watch.go

```diff
@@ -0,0 +1,59 @@
+package fswatcher
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// Watch for changes to a directory on the filesystem and sends a notification to eventCh every time a file in the folder is changed.
+// Although it's possible to watch for individual files, that's not recommended; watch for the file's parent folder instead.
+// Note that changes are batched for 0.5 seconds before notifications are sent
+func Watch(ctx context.Context, dir string, eventCh chan<- struct{}) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf("failed to create watcher: %w", err)
+	}
+	defer watcher.Close()
+
+	err = watcher.Add(dir)
+	if err != nil {
+		return fmt.Errorf("watcher error: %w", err)
+	}
+
+	batchCh := make(chan struct{}, 1)
+	defer close(batchCh)
+
+	for {
+		select {
+		// Watch for events
+		case event := <-watcher.Events:
+			if event.Op&fsnotify.Create == fsnotify.Create ||
+				event.Op&fsnotify.Write == fsnotify.Write {
```

Comment:
Should we also add support for `Remove`, `Rename` and `Chmod`? If not, we should add it to the documentation comment.

Reply:
They were not in the code that was originally in the runtime and for the way it's used in the runtime, I don't think they're needed.

Reply:
@ItalyPaleAle I know that this PR is just bringing what is existing in runtime into kit repo. But I believe it will be better if we can have some form of struct detailing the event being sent back in the `eventCh`. This will later allow for watcher to be extended without extensive modifications to runtime. 
For example watcher is being used in https://github.com/dapr/dapr/pull/5197, I was also thinking we can use it in CLI to watch for file changes and restart dapr or application as a improvement for dev experience --- 

My thought was in the `eventCh`, since its a batch, we can pass in a list of events, essentially whether its a create or a write and the corresponding file details... That would be helpful on the caller side ... 
thoughts ?

Reply:
On other note , this can be a separate change also ...

Reply:
That’s a good suggestion but I wonder if in that case you’d be best just using `fsnotify/fsnotify` instead?

This package is used in the runtime to watch for TLS certs, where there are 2 files that are changed at the same time. The runtime doesn’t really need the complexity of knowing which files have changed, and batching is required.